### PR TITLE
Implement the additional manifests for customized yamls

### LINKED ISSUE #330: Allow users to define additional manifests without wiping the Serving/Eventing core
**Problem**

After defining a custom `manifest` in my KnativeServing object, the entire core Serving landcape was wiped.

I was expecting the operator to deploy my manifest _additionally_ to the core. I think most people do want to have the Knative core when using the operator, so this behaviour was a bit surprising to me, and not that easy to recover from in a live cluster.

**[Persona:](https://github.com/knative/eventing/blob/master/docs/personas.md)** Cluster admins

**Exit Criteria**

Custom manifests can be deployed _on top_ of the core, without having to reference every core manifests from GitHub.

If we don't want to break the current functionality, this will require a new option. I suggest `additionalManifests`.

**Time Estimate (optional):** ?

**Additional context (optional)**

cc. @houshengbo as discussed in Slack

### PR DESCRIPTION
**Pull request by houshengbo**
Fixes #330

## Proposed Changes

* Add the model and implement spec.additionalManifests.

**Release Note**

```release-note

```

### CODE REVIEW

File: pkg/reconciler/common/testdata/kodata/additional-manifests/additional-resource.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: knative-additional
+  labels:
+    serving.knative.dev/release: "v0.16.1"
```
Comment (antoineco) [inline pkg/reconciler/common/testdata/kodata/additional-manifests/additional-resource.yaml:20]: Is this label actually taken into account while reading additional manifests.

If yes, sorry for missing it.

If not, having it there might be misleading, and give the impression that the operator evaluates it, when it doesn't.

Comment (houshengbo) [reply pkg/reconciler/common/testdata/kodata/additional-manifests/additional-resource.yaml:20]: Yes.
If the user does not specify the field spec.version, we do not check if the version of the label matches the spec.version.
If the user specifies the field spec.version, we will validate if the version of the label matches the spec.version.

File: pkg/reconciler/common/releases.go [hunk 2]
```diff
@@ -210,11 +207,32 @@ func componentURL(version string, instance v1alpha1.KComponent) string {
 		url := strings.ReplaceAll(manifest.Url, VersionVariable, version)
 		urls = append(urls, url)
 	}
+
+	manifestPath := strings.Join(urls, COMMA)
```
[the remark this answers is not in the document]
Comment (houshengbo) [reply pkg/reconciler/common/releases.go:211]: The knative operator leverages a library called `github.com/manifestival/manifestival` to retrieve the manifests by the path.
The path can be a single valid link locally or remotely. Also, we can put multiple links together as one string for the library to read all of them. The rule is we need to put COMMA between two adjacent links. This is a defined rule by the library `github.com/manifestival/manifestival`.

To parse a single link, we can use `http://link-1/my-serving.yaml` as the path.
To parse multiple links with one line of code, we can use `http://link-1/my-serving.yaml,http://link-2/my-serving.yaml` as the path.

File: pkg/reconciler/common/releases.go [hunk 3]
```diff
@@ -210,11 +207,32 @@ func componentURL(version string, instance v1alpha1.KComponent) string {
...
+	// If spec.manifests is empty, add the local path
+	if manifestPath == "" {
+		manifestPath = filepath.Join(componentDir(instance), version)
+		if _, err := os.Stat(manifestPath); os.IsNotExist(err) {
+			return ""
+		}
+	}
```
[the remark this answers is not in the document]
Comment (houshengbo) [reply pkg/reconciler/common/releases.go:218]: will do it.

**Review by antoineco — APPROVED** [review]
/lgtm

**Review by antoineco — APPROVED** [review]
/lgtm

**Pull request merged**